modules/vf-graphql-holochain/queries/agent.ts

```diff
@@ -47,5 +60,23 @@ export default (dnaConfig: DNAIdMappings, conductorUri: string) => {
       return ((await readAgent({ address: args.id })).agent) as Person
       // TODO: type check if person or organization and provide error if organization
     }),
+    agents: async (root, args: PagingParams): Promise<AgentConnection> => {
+      let agents = (await readAll(args)) as AgentConnection
+      agents.edges = agents.edges.map((agentEdge) => {
+        agentEdge.node['__typename'] = 'Person'
+        return agentEdge
+      })
+      return agents
+    },
+    organizations: async (root, args: PagingParams): Promise<OrganizationConnection> => {
+      let agents = await readAll(args)
+      agents.edges = agents.edges.filter((agentEdge) => agentEdge.node.agentType === 'Organization')
+      return (agents as OrganizationConnection)
+    },
+    people: async (root, args: PagingParams): Promise<PersonConnection> => {
+      let agents = await readAll(args)
+      agents.edges = agents.edges.filter((agentEdge) => agentEdge.node.agentType === 'Person')
+      return (agents as PersonConnection)
+    },
```

Comment:
Just noting that filtering at the gql layer means that the paging params won't be completely accurate to the returned results (e.g. ask for the latest 10 people, but backend returns latest 10 agents, and then filters organizations out of that, possibly returning less than 10)

Reply:
check out https://github.com/h-REA/hREA/issues/327#issuecomment-1174644140 (;